jobbergate-core/jobbergate_core/auth/core.py

```diff
@@ -0,0 +1,316 @@
+"""
+Utilities for handling authentication in the Jobbergate system.
+"""
+import time
+from collections import namedtuple
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Dict, cast
+
+import httpx
+from loguru import logger
+
+from .exceptions import AuthenticationError, TokenError
+from .token import Token, TokenType
+
+
+_LoginInformation = namedtuple(
+    "_LoginInformation",
+    ["verification_url", "wait_interval", "device_code"],
+)
+
+
+@dataclass
+class JobbergateAuth:
+    """
+    High-level class used to handle authentication in Jobbergate.
+
+    After an instance of this class is created, it can be used to authenticate requests
+    from both `requests`_  and `httpx`_ packages by passing it to the ``auth``
+    parameter on the request (see examples bellow).
```

Comment:
```suggestion
    parameter on the request (see examples below).
```

Reply:
solved in [7d94777](https://github.com/omnivector-solutions/jobbergate/pull/228/commits/7d9477776ec7fce9deea844be884ddcd06bef054)

Reply:
[705826b](https://github.com/omnivector-solutions/jobbergate/pull/228/commits/705826b5482c89b424aa285a0cd38a4cc6f428e9)